src/town_gui.cpp

```diff
@@ -1548,7 +1544,7 @@ class HousePickerCallbacks : public PickerCallbacks {
 			if (*it == 0) continue;
 			HouseID house = static_cast<HouseID>(std::distance(id_count.begin(), it));
 			const HouseSpec *hs = HouseSpec::Get(house);
-			int class_index = FindFirstBit(hs->building_availability & HZ_ZONALL);
+			int class_index = FindFirstBit((hs->building_availability & HZ_ZONE_ALL).base());
```

Comment:
Since the 0 offset is added in `GetHouseZoneFromClassId`, it should be subtracted here as well. 
```suggestion
			int class_index = FindFirstBit((hs->building_availability & HZ_ZONE_ALL).base()) - to_underlying(HouseZone::TownEdge);
```